stdpopsim/catalog/HomSap/__init__.py

```diff
@@ -1650,3 +1650,175 @@ def _AJ():
+    T_CH_JP = int(9000 / generation_time)  # time of Han/Japanese split
+
+    # also
+    r_AF = 0  # YRI and B growth rate
+
+    # A and YRI
+    pop0 = msprime.PopulationConfiguration(
+        initial_size=N_AF * math.exp(r_AF * T_AF),
+        growth_rate=r_AF,
+        metadata=populations[0].asdict(),
+    )
+
+    # B and CEU
+    pop1 = msprime.PopulationConfiguration(
+        initial_size=N_EU0 * math.exp(r_EU * T_EU_AS),
+        growth_rate=r_EU,
+        metadata=populations[1].asdict(),
+    )
+
+    # Asian and CHB
+    pop2 = msprime.PopulationConfiguration(
+        initial_size=N_AS0 * math.exp(r_AS * T_EU_AS),
+        growth_rate=r_AS,
+        metadata=populations[2].asdict(),
+    )
+
+    # JPT
+    pop3 = msprime.PopulationConfiguration(
+        initial_size=N_JP0 * math.exp(r_JP * T_CH_JP),
+        growth_rate=r_JP,
+        metadata=populations[3].asdict(),
+    )
+
+    population_configurations = [pop0, pop1, pop2, pop3]
+
+    migration_matrix = [
+        [0, m_AF_EU, m_AF_AS, 0],
+        [m_AF_EU, 0, m_EU_AS, 0],
+        [m_AF_AS, m_EU_AS, 0, m_CH_JP],
+        [0, 0, m_CH_JP, 0],
+    ]
+
+    # Demographic events from most recent -> oldest
+
+    # JPT splits from CHB
+    JPT_event = [
+        # seg migration rate JPT <--> CHB to zero
+        msprime.MigrationRateChange(time=T_CH_JP, rate=0.0, matrix_index=(2, 3)),
+        msprime.MigrationRateChange(time=T_CH_JP, rate=0.0, matrix_index=(3, 2)),
+        # JPT splits from CHB
+        msprime.MassMigration(time=T_CH_JP, source=3, destination=2, proportion=1.0),
+        # set growth rate in JPT to zero
+        msprime.PopulationParametersChange(
+            time=T_CH_JP + 0.0001, growth_rate=0.0, population_id=3
+        ),
+    ]
+
+    # CEU and CHB merge into B
+    AS_event = [
+        # set migration rates to zero
+        msprime.MigrationRateChange(time=T_EU_AS, rate=0.0),
+        # CHB splits from CEU
+        msprime.MassMigration(time=T_EU_AS, source=2, destination=1, proportion=1.0),
+        # change size of B to N_B
+        # set growth rates in B and CHB to zero,
+        msprime.PopulationParametersChange(
+            time=T_EU_AS + 0.0001, initial_size=N_B, growth_rate=0.0, population_id=1
+        ),
+        msprime.PopulationParametersChange(
+            time=T_EU_AS + 0.0001, growth_rate=0.0, population_id=2
+        ),
+        # set up migration between YRI and B
+        msprime.MigrationRateChange(
+            time=T_EU_AS + 0.0001, rate=m_AF_B, matrix_index=(0, 1)
+        ),
+        msprime.MigrationRateChange(
+            time=T_EU_AS + 0.0001, rate=m_AF_B, matrix_index=(1, 0)
+        ),
+    ]
+
+    # Out-of-Africa event
+    EU_event = [
+        # B splits from YRI
+        msprime.MigrationRateChange(time=T_B, rate=0.0),
+        msprime.MassMigration(time=T_B, source=1, destination=0, proportion=1.0),
+    ]
+
+    # Change size of YRI/African population
+    AF_event = [
```

Comment:
I think this should have `time=T_AF` instead of `T_B`.

Reply:
Good catch!